library/core/src/ptr/const_ptr.rs

```diff
@@ -906,6 +906,41 @@ impl<T> *const [T] {
         // SAFETY: the caller ensures that `self` is dereferencable and `index` in-bounds.
         unsafe { index.get_unchecked(self) }
     }
+
+    /// Returns the pointee as a shared, uninitialized slice.
+    ///
+    /// # Safety
+    ///
+    /// Behavior is undefined if any of the following conditions are violated:
+    ///
+    /// * The pointer must be [valid] for reads for `ptr.len() * mem::size_of::<T>()` many bytes,
+    ///   and it must be properly aligned. This means in particular:
+    ///
+    ///     * The entire memory range of this slice must be contained within a single allocated object!
+    ///       Slices can never span across multiple allocated objects.
+    ///     * The pointer must be non-null and aligned even for zero-length slices. One
+    ///       reason for this is that enum layout optimizations may rely on references
+    ///       (including slices of any length) being aligned and non-null to distinguish
+    ///       them from other data. You can obtain a pointer that is usable as `data`
+    ///       for zero-length slices using [`NonNull::dangling()`].
+    ///
+    /// * The memory referenced by the returned slice must not be mutated for the duration
+    ///   lifetime `'a`, except inside an `UnsafeCell`.
+    ///
+    /// * The total size `ptr.len() * mem::size_of::<T>()` of the slice must be no larger than `isize::MAX`.
+    ///   See the safety documentation of [`pointer::offset`].
+    ///
+    /// See also [`slice::from_raw_parts`].
+    ///
+    /// [valid]: ../index.html#safety
+    /// [`NonNull::dangling()`]: NonNull::dangling
+    /// [`pointer::offset`]: ../primitive.pointer.html#method.offset
+    #[inline]
+    #[unstable(feature = "uninit_ptr_slice", issue = "none")]
+    pub unsafe fn as_uninit_slice<'a>(self) -> &'a [MaybeUninit<T>] {
```

Comment:
Please note in the tracking issue that the naming is different from the usual raw ptr methods which indicate mutability -- like `from_raw_parts_mut` or `as_mut`. I don't have a strong opinion either way, but the libs team might. ;)

Reply:
I could rename the feature to `slice_ptr_uninit` which follows the naming scheme of #74265.

Reply:
I don't think anyone cares about feature names, but sure.^^
I mean the names of the methods.

Reply:
Oh sure :smile: 

Well, currently the term "uninit" is always prepended to the type: https://doc.rust-lang.org/nightly/core/index.html?search=uninit

Reply:
I was talking about the fact that the const and mut methods have the same name, which they usually do not.

Reply:
Oh :D
I'll change this.